packages/@aws-cdk/cloud-assembly-schema/lib/manifest.ts

```diff
@@ -97,9 +102,7 @@ export class Manifest {
 
     // first validate the version should be accepted.
     if (semver.gt(actual, maxSupported)) {
-      // we use a well known error prefix so that the CLI can identify this specific error
-      // and print some more context to the user.
-      throw new Error(`${VERSION_MISMATCH}: Maximum schema version supported is ${maxSupported}, but found ${actual}`);
+      throw new VersionMismatchError();
```

Comment:
I mourn the loss of specific error information, only to be replaced by contentless generic messaging.

Reply:
This wasn't merged yet, so we can hold off on the mourning :)

When this error is thrown, the action item for the user is  "Upgrade the CLI". The cloudAssembly scheme is not a well enough known concept, it's not something that most end users are aware of, so in this case the extra information is just confusing. When a user have a cli version `1.69` and is using version `1.72` of the framework, throwing:

 `(Cloud assembly schema version mismatch: Maximum schema version supported is 6.0.0, but found 7.0.0)`

Ends up confusing users who are only familiar with version `1.69`/`1.72`.